.github/workflows/test.yml

```diff
@@ -13,7 +13,7 @@ jobs:
       - uses: actions/checkout@v1
       - run: |
           npm install
-          npm run all
+          GITHUB_ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm run all
```

Comment:
Does still not work, lets try it with a env option for this step see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsenv

Reply:
Perhaps you also have to change the env var name, not sure.

Reply:
I've just added the env option and still failed

Reply:
https://github.com/GsActions/commit-message-checker/blob/1a845b6ecf28d47368ddeb8b41021ca416056de3/.github/workflows/test.yml#L14-L18

Reply:
is this right?

Reply:
Yeah, saw it. As said perhaps you also have to rename the env var

Reply:
ok, I will rename it

Reply:
It would try it now with `ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}`

Reply:
ACCESS_TOKEN is still undefined in `input-test.test.js`

Reply:
But I was right not using GITHUB_ see https://docs.github.com/en/actions/reference/environment-variables#naming-conventions-for-environment-variables

Reply:
https://github.com/ItaloYeltsin/commit-message-checker/blob/b84ee10a68c2d69efd35e01c52be9cdb6b71cd61/.github/workflows/test.yml#L10-L18

Reply:
https://github.com/ItaloYeltsin/commit-message-checker/blob/b84ee10a68c2d69efd35e01c52be9cdb6b71cd61/.github/workflows/test.yml#L10-L18

Now the action stoped without any result.

Reply:
Yep, I guess the definition is wrong. Try this please:
```
      - uses: actions/checkout@v1
      - name: Test
        env: 
          ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
        run: |
...
```

Reply:
Or better use the name "Build & Test"

Reply:
it didn't work =(

Reply:
```yaml
     - name: Build & Test
          npm install	        run: |
          npm run all	          ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm install
          ACCESS_TOKEN=${{ secrets.ACCESS_TOKEN }} npm run all
````
There is something wrong with passing env variables to npm install, not even this is working

Reply:
I've created a new secret "MYSECRET" which you can use to play around and also expose to logs etc, it simply contains "mygreatsecret". I currently have no idea what's the problem here, according the GH Actions help this should work...

Reply:
https://stackoverflow.com/questions/57431439/how-to-set-an-environment-variable-to-be-used-during-npm-install

Reply:
I would start doing testing like done here using echo https://docs.github.com/en/actions/reference/environment-variables#about-environment-variables. Therefor I created the new secret. But anyway the token has read access only, so you also could use the real token...

Reply:
![grafik](https://user-images.githubusercontent.com/25326036/93520576-fc9a3d80-f92e-11ea-9ce2-cac863bec880.png)

Reply:
Have a look here https://docs.github.com/en/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

You should be able to use GITHUB_TOKEN instead, the only passed secret for PRs from forks.

Reply:
@ItaloYeltsin Have a look at my last comment

Reply:
And please use the env option if possible, looks nicer...

Reply:
Oh no, you can just access it in the tests, no need to define something, it's already there :)

Reply:
ok

Reply:
So, should I change it to GITHUB_TOKEN?

Reply:
I already did the changes and we got a green build now :)

Reply:
Nice, it seems it worked!